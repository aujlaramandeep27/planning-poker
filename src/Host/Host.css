.ticket-card, .ticket-button {
    background: #2359fa;
    background-color: #2359fa;
}

.ticket-card:hover, .uk-button-secondary:hover, .add-card:hover {
    background-color: #b1bafc;
}
