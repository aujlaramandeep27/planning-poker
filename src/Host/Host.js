import React from 'react';
import { Link } from 'react-router-dom';
import { viewAddStory, viewMeeting, viewMeetings } from '../Navigation/route-actions';
import { Page } from '../Common/Header';
import { connect } from 'react-redux';

export function Host({ currentMeeting, addStory, goToMeetings, estimate }) {

  let alphabets = 'ABCD';

  return (
    <Page title='Host' onBack={ goToMeetings }>
      <button className="uk-button uk-button-primary uk-button-small uk-position-small uk-position-top-right"
              onClick={ () => estimate(currentMeeting._id) }>
        Estimate
      </button>

      { Array.from(alphabets).map((letter, i) =>
        <div key={ i } className="uk-card uk-card-secondary uk-card-body uk-margin pp-button">
          <Link to={ { pathname: '/story/' + letter, query: { storyId: letter } } }>
            <div className="uk-card-badge uk-label">Points: 3</div>
            <h3 className="uk-card-title">Story { letter }</h3>
          </Link>
        </div>
      )
      }

      <div className="uk-card uk-card-secondary uk-card-body pp-secondary-button" onClick={ addStory }>
        <h3 className="uk-card-title">+</h3>
      </div>
    </Page>
  );
}

function mapStateToProps(state) {
    return {
        currentMeeting: state.currentMeeting
    };
}

function mapDispatchToProps(dispatch) {
  return {
    goToMeetings: () => dispatch(viewMeetings()),
    addStory: () => dispatch(viewAddStory()),
    estimate: (meetingId) => dispatch(viewMeeting(meetingId))
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Host);
