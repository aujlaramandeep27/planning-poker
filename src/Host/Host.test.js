import { Host } from './Host';
import { noop } from 'lodash';

it('Host renders correctly', () => {
  expect(Host({ addStory: noop, estimate: noop })).toMatchSnapshot();
});
