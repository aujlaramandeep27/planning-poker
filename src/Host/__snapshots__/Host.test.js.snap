// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Host renders correctly 1`] = `
<React.Fragment>
  <div>
    <div
      className="uk-container uk-text-center@m"
    >
      <h1
        className="uk-heading-divider uk-margin-top"
      >
        Host
      </h1>
      <div
        className="uk-align-center uk-width-1-2@m"
      >
        <a
          href="#/meetings/"
        >
          <span
            className="uk-position-large uk-position-top-left"
            uk-icon="icon: arrow-left; ratio: 3"
          />
        </a>
        <Link
          to="/estimate"
        >
          <button
            className="uk-button uk-button-primary uk-button-small uk-position-small uk-position-top-right"
          >
            Estimate
          </button>
        </Link>
        <div
          className="uk-card uk-card-secondary uk-card-body uk-margin pp-button"
        >
          <Link
            to={
              Object {
                "pathname": "/story/A",
                "query": Object {
                  "storyId": "A",
                },
              }
            }
          >
            <div
              className="uk-card-badge uk-label"
            >
              Points: 3
            </div>
            <h3
              className="uk-card-title"
            >
              Story 
              A
            </h3>
          </Link>
        </div>
        <div
          className="uk-card uk-card-secondary uk-card-body uk-margin pp-button"
        >
          <Link
            to={
              Object {
                "pathname": "/story/B",
                "query": Object {
                  "storyId": "B",
                },
              }
            }
          >
            <div
              className="uk-card-badge uk-label"
            >
              Points: 3
            </div>
            <h3
              className="uk-card-title"
            >
              Story 
              B
            </h3>
          </Link>
        </div>
        <div
          className="uk-card uk-card-secondary uk-card-body uk-margin pp-button"
        >
          <Link
            to={
              Object {
                "pathname": "/story/C",
                "query": Object {
                  "storyId": "C",
                },
              }
            }
          >
            <div
              className="uk-card-badge uk-label"
            >
              Points: 3
            </div>
            <h3
              className="uk-card-title"
            >
              Story 
              C
            </h3>
          </Link>
        </div>
        <div
          className="uk-card uk-card-secondary uk-card-body uk-margin pp-button"
        >
          <Link
            to={
              Object {
                "pathname": "/story/D",
                "query": Object {
                  "storyId": "D",
                },
              }
            }
          >
            <div
              className="uk-card-badge uk-label"
            >
              Points: 3
            </div>
            <h3
              className="uk-card-title"
            >
              Story 
              D
            </h3>
          </Link>
        </div>
        <Link
          to="/tickets/add"
        >
          <div
            className="uk-card uk-card-secondary uk-card-body pp-secondary-button"
          >
            <h3
              className="uk-card-title"
            >
              +
            </h3>
          </div>
        </Link>
      </div>
    </div>
  </div>
</React.Fragment>
`;
